src/main/java/lotto/domain/result/Rank.java

```diff
@@ -7,31 +7,46 @@
 public enum Rank {
 
     NONE(0, Money.of(0)),
-    FOURTH(3, Money.of(5_000)),
-    THIRD(4, Money.of(50_000)),
-    SECOND(5, Money.of(1_500_000)),
-    FIRST(6, Money.of(2_000_000_000))
+    FIFTH(3, Money.of(5_000)),
+    FOURTH(4, Money.of(50_000)),
+    THIRD(5, Money.of(1_500_000)),
+    SECOND(5, Money.of(30_000_000)),
+    FIRST(6, Money.of(2_000_000_000)),
     ;
 
     private final long matchCount;
     private final Money money;
 
+
     Rank(long matchCount, Money money) {
         this.matchCount = matchCount;
         this.money = money;
     }
 
-    public static Rank of(long matchCount) {
-        return getRank(matchCount);
-    }
+    public static Rank findRankBy(long matchCount, boolean bonusMatch) {
+        if (isSecond(matchCount, bonusMatch)) {
+            return SECOND;
+        }
 
-    public static Rank getRank(long matchCount) {
         return Arrays.stream(values())
-                .filter(r -> r.matchCount == matchCount)
+                .filter(Rank::isNotSecond)
+                .filter(r -> isSameMatchCount(matchCount, r))
                 .findFirst()
                 .orElse(NONE);
     }
 
+    private static boolean isSecond(long matchCount, boolean bonusMatch) {
+        return matchCount == 5 && bonusMatch;
+    }
+
+    private static boolean isNotSecond(Rank r) {
+        return !r.equals(SECOND);
```

Comment:
- 객체지향 생활 체조 원칙 중 `규칙 5: 줄여쓰지 않는다(축약 금지).` 항목을 참고해주세요.
- 열거형 상수끼리는 `==`으로도 비교가 가능합니다.
```suggestion
    private static boolean isNotSecond(Rank rank) {
        return rank != SECOND;
```

Reply:
네 감사합니다.